### PR TITLE
add configuration option to allow self signed tls

### PR DESCRIPTION
**Pull request by lfuelling**
Hi @JC5,
this PR fixes issue firefly-iii/firefly-iii#3470 by adding a configuration option to skip the certificate verification of Guzzle.

~~I've never written PHP > 5  (or used Laravel) before so I'm unsure wether this is all correct but I can't find a Dockerfile for testing it (see firefly-iii/firefly-iii#3471).~~

I was able to test it, unfortunately something still does not work.

### CODE REVIEW

File: app/Services/Spectre/Request/Request.php [hunk 1]
```diff
@@ -242,7 +242,7 @@ protected function sendSignedSpectrePost(array $data): array
 
         Log::debug('Final headers for spectre signed POST request:', $headers);
         try {
-            $client = new Client;
+            $client = $this->getClient();
```
Comment (lfuelling) [inline app/Services/Spectre/Request/Request.php:245]: For example I'm not sure if there is a specific reason that `getClient()` wasn't used here before.

**Comment by JC5** [comment]
From the place where you change the code, it seems that your connection to Spectre fails. That's not a self-signed certificate and that's NOT something the certificate verification should be skipped.

**Comment by lfuelling** [comment]
Oh so `Request` only connects to Spectre? 

Then no wonder that it's not working. 🤦 

But the issue I'm having and am trying to fix really is the connection to Firefly which is using a self signed certificate.

**Comment by lfuelling** [comment]
Hi again @JC5,

it seems like I have to change code in the `GrumpyDictator\FFIIIApiSupport` package. 
Could you please point me to the sources of this? I was unable to find such a  repository in the firefly-ii org.

**Comment by JC5** [comment]
https://github.com/JC5/api-support-classes

If you're going to submit a change to this code, make sure it's an option in maybe the constructor or something. Let me know what you have in mind!

**Comment by JC5** [comment]
Please reopen this PR on the develop branch, I've just noticed this. And drop the changes to the gitignore file. Those should go into your own global gitignore file.

**Comment by lfuelling** [comment]
Closing it wasn't even necessary :o

**Pull request merged**